File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 1
    routes.sort()
    start, end = routes[0]
    for s,e in routes:
        if(start<=s <=end):
            start = max(start, s)
            end = min(end, e)
        else:
            answer+=1
            start = s
            end =e
        print(s,e, start, end, answer)
    print(routes)
    return answer

File: 프로그래머스/3/42884. 단속카메라/단속카메라.js
function solution(routes) {
    var answer = 1;
    routes.sort((a,b)=>a[0]-b[0])
    console.log(routes)
    let [start,end] = [routes[0][0],routes[0][1]]
    for(let [s,e] of routes){
        if(start<=s && s<=end ){
            start = Math.max(start, s)
            end = Math.min(end,e)      
            // console.log(start, end, s, e, answer, "**")
        }else{
            answer+=1
            start = s
            end=e
            // console.log(start, end, s, e, answer, "****")
        }
        
    }
    return answer;
}